src/webapp/pages/settings/SettingsPage.tsx

```diff
@@ -0,0 +1,204 @@
+import i18n from "@dhis2/d2-i18n";
+import _ from "lodash";
+import { TableAction, RowConfig, ObjectsTable, TableState, useSnackbar } from "@eyeseetea/d2-ui-components";
+import { Button } from "@dhis2/ui";
+import React, { useCallback, useState, useMemo, useEffect } from "react";
+import { PageHeader } from "../../components/page-header/PageHeader";
+import { useGoBack } from "../../hooks/useGoBack";
+import { useAppContext } from "../../contexts/app-context";
+import { ListMetadataResponse, ListOptions } from "../../../domain/repositories/MetadataRepository";
+import { displayName, MetadataItem, MetadataModel } from "../../../domain/entities/MetadataItem";
+import Dropdown, { DropdownOption } from "../../components/dropdown/Dropdown";
+import RemoveCircleOutlineIcon from "@material-ui/icons/RemoveCircleOutline";
+import AddCircleOutlineIcon from "@material-ui/icons/AddCircleOutline";
+
+export const SettingsPage: React.FC = () => {
```

Comment:
Refactor: this component has a lot of logic, so it needs to be splitted into a hook that contains all the presentation logic and that the component only contains render logic.